aten/src/ATen/core/typeid.h

```diff
@@ -391,53 +391,26 @@ inline bool operator!=(const TypeMeta& lhs, const TypeMeta& rhs) noexcept {
  *
  * NOTE: the macro needs to be invoked in ::caffe2 namespace
  */
-// Implementation note: in MSVC, we will need to prepend the AT_CORE_API
-// keyword in order to get things compiled properly. in Linux, gcc seems to
-// create attribute ignored error for explicit template instantiations, see
-//   http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/p0537r0.html
-//   https://gcc.gnu.org/bugzilla/show_bug.cgi?id=51930
-// and as a result, we define these two macros slightly differently.
```

Comment:
@xkszltl sorry to mention this, but we need to revert the diff due to an internal build error, so I'd like to update you here :) Basically, internally we are turning on "-Werror=attribute" and the removal of the ifdef below made build fail on Linux. We are going to add this build flag to the oss build as well to bring build settings consistent.

Thanks so much for the fix though! If you don't mind, could you do a follow up fix for this and we can re-merge?